templates/puppet.spec

```diff
@@ -0,0 +1,52 @@
+%{!?upstream_version: %global upstream_version %{version}%{?milestone}}
+{% if metadata.from_puppetlabs -%}
+%%define upstream_name {{ metadata.puppetlabs_name }}
+{% endif %}
+Name:                   {{ metadata.name }}
+Version:                XXX
+Release:                XXX
+Summary:                {{ metadata.summary }}
+License:                {{ metadata.license }}
+
+URL:                    {{ metadata.project_page }}
+
+Source0:                {{ metadata.source0 }}
+
+BuildArch:              noarch
+
+{% for dep in metadata.dependencies -%}
+Requires:               {{ dep.name }}
+{% endfor %}
+Requires:               puppet >= 2.7.0
+
+%description
+{{ metadata.description }}
+
+%prep
+%setup -q -n {{ '%{'+metadata.upstream_name+'}-%{upstream_version}' }}
+
+find . -type f -name ".*" -exec rm {} +
+find . -size 0 -exec rm {} +
+find . \( -name "*.pl" -o -name "*.sh"  \) -exec chmod +x {} +
+find . \( -name "*.pp" -o -name "*.py"  \) -exec chmod -x {} +
+find . \( -name "*.rb" -o -name "*.erb" \) -exec chmod -x {} +
+find . \( -name spec -o -name ext \) | xargs rm -rf
+
+%build
+
+
+%install
+rm -rf %{buildroot}
+install -d -m 0755 %{buildroot}/%{_datadir}/openstack-puppet/modules/{{ metadata.project }}/
+cp -rp * %{buildroot}/%{_datadir}/openstack-puppet/modules/{{ metadata.project }}/
+{% if "nova" == metadata.project -%}
```

Comment:
Agree, basically the goal of this patch is a straight port of the existing embedded logic into the template, with the intent to remove or abstract things like this eventually.  I am not yet sure the best way to do it, but I would propose dealing with these things in an iterative fashion, as this is already a somewhat massive patch (I prefer much smaller ones, but could not really pull that off for this go 'round).